project/recursive_sorting.py

```diff
@@ -36,20 +42,45 @@ def merge_in_place(arr, start, mid, end):
 
     return arr
 
-def merge_sort_in_place(arr, l, r): 
+
+def merge_sort_in_place(arr, l, r):
     # TO-DO
 
     return arr
 
 
 # TO-DO: implement the Quick Sort function below USING RECURSION
-def quick_sort( arr, low, high ):
-
+def quick_sort(arr, low, high):
```

Comment:
it works! jaja however you're moving the pivot before you go through the entire array to then find it's place. It looks like an optimization! But I just wanted to point out that it's different than what we're seeing in class.
Usually, you'd ignore the value before the pivot (you're tracking it on line 63) and simply compare the current index to the pivot, swapping if it's lower and advancing the "swap index" (index_low) in your code. this leaves you with unsorted arrays on either side of the pivot, but with the knowledge that everything on the left is smaller and everything on the right is larger. 
I ran your code with print statements, and it doesn't seem obvious that you get that effect. 
That said though, the code still runs fine and there's something to be said that you figured out a different way of doing it.

Reply:
I was having trouble staying online during lecture that day, so I missed a lot of the explanation of the quick-sort algorithm. I relied on the Wikipedia article and several sorting videos on Youtube to understand the algorithm, and decided to base my implementation on this diagram:
https://upload.wikimedia.org/wikipedia/commons/thumb/a/af/Quicksort-diagram.svg/200px-Quicksort-diagram.svg.png